src/include/txn.h

```diff
@@ -288,6 +288,13 @@ struct __wt_txn {
     /* Read updates committed as of this timestamp. */
     wt_timestamp_t read_timestamp;
 
+    /*
+     * Set to the first read timestamp used in the transaction. As part of our history store
+     * mechanism, we can move the read timestamp forward so we need to keep track of the original
+     * read timestamp to know what history should be pinned in front of oldest.
+     */
+    wt_timestamp_t first_read_timestamp;
```

Comment:
The reason I didn't modify the querying is because `read_timestamp != first_read_timestamp` for only a small window while we're doing a history store modify read before getting restored.

So we'll never be able to query the session at a time where they're not equal.

Reply:
> So we'll never be able to query the session at a time where they're not equal.

I get that, but I think of it as future-proofing. This new field is the "truth" for everything but the current search, right? If that statement is correct, that should be reflected in the code?

Reply:
487e5f3